orchestrator-salt/src/main/resources/salt-common/salt/monitoring/request-signer.sls

```diff
@@ -0,0 +1,46 @@
+{%- from 'monitoring/settings.sls' import monitoring with context %}
+{%- if monitoring.enabled and monitoring.requestSignerEnabled and monitoring.databusEnabled %}
+{% set request_signer_installed = salt['file.directory_exists' ]('/opt/cdp-request-signer/bin') %}
+{%- if not request_signer_installed %}
+install_request_signer:
+  cmd.run:
+    - name: "yum install -y cdp-request-signer --disablerepo='*' --enablerepo=cdp-infra-tools; exit 0"
+    - onlyif: "test -f /etc/yum.repos.d/cdp-infra-tools.repo && ! rpm -q cdp-request-signer"
+{%- endif %}
```

Comment:
if request signer is not installed and infra repo is not there, the state after this would still try to run. Either they should depend on this state or put behind an if statement. what do you think?

Reply:
maybe it's easier if this command would fail (so we expecting monitoring will work but cannot install a component)

Reply:
but `exit 0` would cause this to not fail

Reply:
yeah, right, sorry missed that